File: 2022/test/day_06_test.exs
defmodule Day06Test do
  use ExUnit.Case

  import TestHelper
  import Day06

  test "part1" do
    assert part1(input("day_06_example_1")) == 7
    assert part1(input("day_06_example_2")) == 10
    assert part1(input("day_06")) == 1275
  end

  test "part2" do
    assert part2(input("day_06_example_1")) == 19
    assert part2(input("day_06_example_2")) == 29
    assert part2(input("day_06")) == 3605
  end
end

File: 2024/day06/day06_test.go
package day06

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 41, Part1("input_example_1"))
	assert.Equal(t, 4515, Part1("input"))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, 6, Part2("input_example_1"))
	assert.Equal(t, 5770, Part2("input"))
}
